atomicfu/src/androidNative32BitMain/kotlin/kotlinx/atomicfu/parking/PosixParkingDelegator.kt

```diff
@@ -0,0 +1,59 @@
+package kotlinx.atomicfu.parking
+
+import kotlinx.cinterop.*
+import kotlinx.cinterop.alloc
+import kotlinx.cinterop.pointed
+import kotlinx.cinterop.ptr
+import platform.posix.*
+
+@OptIn(ExperimentalForeignApi::class)
+internal actual object ParkingDelegator {
+    actual fun createRef(): ParkingData {
+        val mut = nativeHeap.alloc<pthread_mutex_t>().ptr
+        val cond = nativeHeap.alloc<pthread_cond_t>().ptr
+        callAndVerifyNative(0)  { pthread_mutex_init(mut, null) }
+        callAndVerifyNative(0)  { pthread_cond_init(cond, null) }
+        return ParkingData(mut, cond)
+    }
+
+    actual fun wait(ref: ParkingData, shouldWait: () -> Boolean){
+        callAndVerifyNative(0)  { pthread_mutex_lock(ref.mut) }
+        if (shouldWait()) callAndVerifyNative(0)  { pthread_cond_wait(ref.cond, ref.mut) }
+        callAndVerifyNative(0)  { pthread_mutex_unlock(ref.mut) }
+    }
+    
+    actual fun timedWait(ref: ParkingData, nanos: Long, shouldWait: () -> Boolean): Unit = memScoped {
+        val ts = alloc<timespec>().ptr
+
+        // Add nanos to current time
+        clock_gettime(CLOCK_REALTIME.toInt(), ts)
+        ts.pointed.tv_sec += nanos.toInt() / 1_000_000_000
+        ts.pointed.tv_nsec += nanos.toInt() % 1_000_000_000
```

Comment:
Here's an incantation that compiles on both 32-bit and 64-bit targets:
```suggestion
        ts.pointed.tv_sec = (ts.pointed.tv_sec + nanos / 1_000_000_000).convert()
        ts.pointed.tv_nsec = (ts.pointed.tv_nsec + nanos % 1_000_000_000).convert()
```
This way, you can remove these custom `androidNative32BitMain`-like source sets and use the source sets provided out of the box like `androidNativeMain` (https://kotlinlang.org/docs/multiplatform-hierarchy.html#see-the-full-hierarchy-template)